src/br/com/zup/nossacasacodigo/author/Author.java

```diff
@@ -39,4 +41,56 @@ public String getDescription() {
 	public Date getCreatedAt() {
 		return this.createdAt;
 	}
+	
+	//Valida��es
+	public static void emailValidator(String email) {
+		isEmpty(email);
+
+		boolean isEmailValid = false; 
+		if (email != null && email.length() > 0) {
+			//Determinamos uma express�o padr�o que determina um e-mail v�lido atrav�s da String expression
+			String expression = "^[\\w\\.-]+@([\\w\\-]+\\.)+[A-Z]{2,4}$";
+			//Definimos nossa express�o como um padr�o e a tornamos sens�vel a letras maiusculas
+			Pattern pattern = Pattern.compile(expression, Pattern.CASE_INSENSITIVE);
+			//verificamos se o e-mail inserido obedece ao padr�o determinado
+			Matcher matcher = pattern.matcher(email);
+			if (matcher.matches()) {
+				isEmailValid = true;
+			}
+		}
+		
+		if(isEmailValid == false) {
+			throw new IllegalArgumentException();
```

Comment:
_Deixamos pistas que facilitem o uso do código onde não conseguimos resolver com compilação._

Tudo bem a exceção ficar sem uma mensagem de erro aqui :D. Mas é sempre legal deixar uma mensagem para se, caso a exceção acontecer em tempo de execução a pessoa que tomou o erro tentar identificar a causa de uma forma mais rápida

Reply:
Oi Larissa, aqui é bem o que Yuri falou. A exception denota um bug na aplicação e ela precisa carregar algum tipo de dica que ajude a pessoa a resolver. Se essa IllegalArgumentException for lançada, a pessoa vai bater o olho na pilha de erros e não vai conseguir entender saber nada sobre o problema além da linha.